lib/src/migrators/division.dart

```diff
@@ -187,6 +193,17 @@ class _DivisionMigrationVisitor extends MigrationVisitor {
     }, isDivisionAllowed: true);
   }
 
+  /// Sets [_negatedParenthesized] to true when about to visit a negated
+  /// parenthesized expression.
+  @override
+  void visitUnaryOperationExpression(UnaryOperationExpression node) {
+    if (node.operator == UnaryOperator.minus &&
+        node.operand is ParenthesizedExpression) {
+      _negatedParenthesized = true;
```

Comment:
You're effectively using this as an out-of-band way to pass an argument to `visitParenthesizedExpression()`... but what if instead you added a `{bool negated = false}` arg to `visitParenthesizedExpression()` and invoked it explicitly here?

Reply:
Good point. Done.